### ISSUE TITLE
Formatter: Unicode width is too high

### ISSUE THREAD

**Issue by konstin** [astral-sh/ruff#6499]
The following snippet fits visually in pycharm and is kept in this layout by pycharm
```python
function_call(
    "aaaaaaaaaaaaaaaaaaaa", "我隻氣墊船裝滿晒鱔.txt", "मेरी मँडराने वाली नाव सर्पमीनों से भरी ह"
)
```
ruff formats it as
```python
function_call(
    "aaaaaaaaaaaaaaaaaaaa",
    "我隻氣墊船裝滿晒鱔.txt",
    "मेरी मँडराने वाली नाव सर्पमीनों से भरी ह",
)
```
We likely compute the unicode width of the string too high so that we assume the line is too long

**Comment by MichaReiser** [comment]
How did you measure the width in Pycharm? I understand that Pycharms cursor position (row:column) is character based and not width based.

**Comment by konstin** [comment]
Really just visually:

![Untitled](https://github.com/astral-sh/ruff/assets/6826232/9fc1626f-b581-400f-9d03-83a87c4052e9)

The chinese characters line up with two latin characters, the hindi doesn't.

I don't know by which rules which tool measures, but we should try to match what black does

**Comment by MichaReiser** [comment]
Black migrates to use unicode-width in preview style, but only for strings (not identifiers).

**Comment by konstin** [comment]
black does also break in preview mode. I guess it makes more sense by the unicode text width algorithm but it's also odd because it's different from the actual text rendering
